src/convert.jl

```diff
@@ -387,6 +387,72 @@ function convert(::Type{Interval}, S::LazySet{N}) where {N<:Real}
     return Interval(-ρ(N[-1], S), ρ(N[1], S))
 end
 
+"""
+    convert(::Type{Hyperrectangle}, cpa::CartesianProductArray{N, <:AbstractHyperrectangle{N}}) where {N<:Real}
+
+Converts the cartesian product of a finite number of hyperrectangular sets to
+a single hyperrectangle.
+
+### Input
+
+- `S`              -- cartesian product array of hyperrectangular set 
+- `Hyperrectangle` -- type used for dispatch
+
+### Output
+
+A hyperrectangle.
+
+### Algorithm
+
+This implementation uses the `center` and `radius_hyperrectangle` methods of
+`AbstractHyperrectangle`.
+"""
+function convert(::Type{Hyperrectangle}, cpa::CartesianProductArray{N, <:AbstractHyperrectangle{N}}) where {N<:Real}
+     n = dim(cpa)
+     c = Vector{N}(undef, n)
+     r = Vector{N}(undef, n)
+     i = 1
+     @inbounds for block_set in array(cpa)
+         j = i + dim(block_set) - 1
+         c[i:j] = center(block_set)
+         r[i:j] = radius_hyperrectangle(block_set)
+         i = j + 1
+     end
+     return Hyperrectangle(c, r)
+end
+
+"""
+    convert(::Type{Hyperrectangle}, cpa::CartesianProductArray{N, <:Interval{N}}) where {N<:Real}
+
+Converts the cartesian product of a finite number of intervals to a single
+hyperrectangle.
+
+### Input
+
+- `S`              -- cartesian product array of intervals 
+- `Hyperrectangle` -- type used for dispatch
+
+### Output
+
+A hyperrectangle.
+
+### Algorithm
+
+This implementation uses the `min` and `max` methods of `Interval` to reduce
+the allocatons and improve performance (see LazySets#1143).
+"""
+function convert(::Type{Hyperrectangle}, cpa::CartesianProductArray{N, <:Interval{N}}) where {N<:Real}
```

Comment:
In think the `<:` in `<:Interval` is not needed since `Interval` does not have subtypes.

Reply:
with the signature

```julia
function convert(::Type{Hyperrectangle}, cpa::CartesianProductArray{N, <:Interval{N}}) where {N<:Real}
...
```
i get `ERROR: LoadError: LoadError: syntax: invalid "::" syntax`

Reply:
Well, yes, remove the `::` :smile:
You also write `N` and not `::N` there.